src/Joblify.Core.Api/Login/LoginController.cs

```diff
@@ -1,10 +1,35 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using System.Threading.Tasks;
+using Joblify.Core.Login;
+using Joblify.Core.Login.Dto;
+using Microsoft.AspNetCore.Mvc;
 
 namespace Joblify.Core.Api.Login
 {
-    [Route("api")]
+    [Route("api/[controller]")]
     public class LoginController : Controller
     {
-        
+        private readonly ILoginService _loginService;
+
+        public LoginController(ILoginService loginService)
+        {
+            _loginService = loginService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Login([FromBody] LoginDto loginDto)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest();
+            }
+
+            var exists = await _loginService.CheckIfUserExists(loginDto);
+            if (exists)
+            {
+                return Ok();
+            }
+            await _loginService.RegisterUser(loginDto);
```

Comment:
what if something here goes wrong? we can track entity state while adding, so we should probably control it